lib/browser/init.js

```diff
@@ -163,8 +163,10 @@ require('./api/protocol')
 const mainStartupScript = packageJson.main || 'index.js'
 
 // Workaround for electron/electron#5050 and electron/electron#9046
-if (process.platform === 'linux' && ['Pantheon', 'Unity:Unity7', 'ubuntu:GNOME', 'pop:GNOME'].includes(process.env.XDG_CURRENT_DESKTOP)) {
-  process.env.XDG_CURRENT_DESKTOP = 'Unity'
+if (process.platform === 'linux') {
+  if (['Pantheon', 'Unity:Unity7', 'pop:GNOME'].includes(process.env.XDG_CURRENT_DESKTOP) || process.env.XDG_CURRENT_DESKTOP.includes('ubuntu')) {
```

Comment:
`process.env.XDG_CURRENT_DESKTOP` is not guaranteed to be defined, we need to handle that case.

If this is going to get bigger I'd like a better structure here.  Something like

```js
const KNOWN_XDG_DESKTOP_VALUES = ['Pantheon', 'Unity:Unity7', 'pop:GNOME']

function currentPlatformSupportsAppIndicator() {
  if (process.platform !== 'linux') return false
  const currentDesktop = process.env.XDG_CURRENT_DESKTOP

  if (!currentDesktop) return false
  if (KNOWN_XDG_DESKTOP_VALUES.includes(currentDesktop)) return true
  if (/ubuntu/ig.test(currentDesktop)) return true

  return false
}

if (currentPlatformSupportsAppIndicator()) {
  process.env.XDG_CURRENT_DESKTOP = 'Unity'
}
```

Reply:
This looks good to me. Good point about undefined `XDG_CURRENT_DESKTOP`, which wasn't already handled by the current implementation.

As your changes could be taken right away, do you want to redo anther MP with yours or me changing that PR to include your code?

Reply:
@didrocks I literally wrote that code on my phone so would be good if you could pull it into this PR and verify it still works 👍

Reply:
Pushed after testing your changes. I added some comments to ensure that the intent is clear. Thanks!